src/hotspot/cpu/x86/x86.ad

```diff
@@ -8615,56 +8629,85 @@ instruct vmuladdaddS2I_reg(vec dst, vec src1, vec src2) %{
 
 // --------------------------------- PopCount --------------------------------------
 
-instruct vpopcountI_popcntd(vec dst, vec src) %{
-  predicate(VM_Version::supports_avx512_vpopcntdq());
+instruct vpopcount_integral_reg_evex(vec dst, vec src) %{
+  predicate((((Matcher::vector_element_basic_type(n->in(1)) == T_INT ||
+               Matcher::vector_element_basic_type(n->in(1)) == T_LONG)) &&
+             VM_Version::supports_avx512_vpopcntdq()) ||
+            (is_subword_type(Matcher::vector_element_basic_type(n->in(1))) &&
+             VM_Version::supports_avx512_bitalg()));
   match(Set dst (PopCountVI src));
-  format %{ "vector_popcount_int $dst, $src\t! vector popcount packedI" %}
+  match(Set dst (PopCountVL src));
+  ins_cost(400);
+  format %{ "vector_popcount_integral_evex $dst, $src" %}
   ins_encode %{
     assert(UsePopCountInstruction, "not enabled");
```

Comment:
UsePopCountInstruction is ambiguous now? subword vs int/long depend on different platform feature so cannot be combined?

Reply:
It a JVM flag, fall back code is still  based on Integer.bitCount and Long.bitCount APIs,  if the flag is turned off we do not intrinsify these scalar APIs.  match_rule_supported which is called through vector inline expander is also sensitive to this flag.

Reply:
match_rule_supported is explicitly checking for avx512popcntdq and avx512bitalg feature. UsePopCountInstruction is only set based on avx512popcntdq feature.